src/pages/index.js

```diff
@@ -0,0 +1,272 @@
+});
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  modal.addEventListener("mousedown", closeModalByOverlay);
+  document.addEventListener("keydown", handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  modal.removeEventListener("mousedown", closeModalByOverlay);
+  document.removeEventListener("keydown", handleEscapeKey);
+}
+
+function closeModalByOverlay(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  };
+}
+
+function handleEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    modals.forEach(modal => closeModal(modal));
+  }
+};
+
+function handleEditModalFormSubmit(evt) {
+  evt.preventDefault();
+  const name = editModalName.value;
+  const about = editModalDescription.value;
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Saving...";
+  api.editUserInfo({ name, about })
+  .then((userInfo) => {
+    profileName.textContent = userInfo.name;
+    profileDescription.textContent = userInfo.about;
+    closeModal(editModal);
+  })
+  .catch(console.error)
+  .finally(() => {
+    submitButton.textContent = "Save";
+  });
+}
+editModalForm.addEventListener("submit", handleEditModalFormSubmit);
+
+function handleAddCardFormSubmit(evt){
+  evt.preventDefault();
+  const initialValues = {
+  name: cardCaptionInput.value,
+  link: cardLinkInput.value };
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Saving...";
+  api.addNewCard(initialValues)
+  .then((card) => {
+  const cardElement = getCardElement(card);
+  cardsList.prepend(cardElement);
+  evt.target.reset();
+  disableButton(cardSubmitBtn, config);
+  closeModal(cardModal);
+})
+.catch(console.error)
+.finally(() => {
+  submitButton.textContent = "Save";
+  });
+}
+cardModalForm.addEventListener("submit", handleAddCardFormSubmit);
+
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(deleteModal);
+}
+
+function handleDeleteCardSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = deleteModalSubmit;
+  submitButton.textContent = "Deleting...";
+  api.deleteCard(selectedCardId)
+  .then(() => {
+    selectedCard.remove();
+    closeModal(deleteModal);
+  })
+  .catch(console.error)
+  .finally(() => {
+    submitButton.textContent = "Delete";
+  });
+}
+deleteModalSubmit.addEventListener("click", handleDeleteCardSubmit);
+
+deleteModalClose.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+deleteModalCancel.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+profileEditButton.addEventListener("click", () => {
+  openModal(editModal);
+  editModalName.value = profileName.textContent.trim();
+  editModalDescription.value = profileDescription.textContent.trim();
+  resetValidation(editModalForm, [editModalName, editModalDescription], {
+    inputErrorClass: "modal__input-error",
+    errorClass: "modal__error",
```

Comment:
Please, use `config` here to avoid duplication of the same selectors

```js
  resetValidation(editModalForm, [editModalName, editModalDescription], config);

```